data-prepper-plugins/mutate-event-processors/src/main/java/org/opensearch/dataprepper/plugins/processor/mutateevent/AddEntryProcessorConfig.java

```diff
@@ -21,6 +21,10 @@ public static class Entry {
         @NotNull
         private Object value;
 
+        @NotEmpty
+        @NotNull
```

Comment:
This value can be null or empty. And with this change, `value` can also be null or empty.

You can use an `@AssertTrue` JSR validation to cover this.

Here is one example, but there are others in the code base:

https://github.com/opensearch-project/data-prepper/blob/8e0fd01180fd1f08ad6ed8288e0b7e437394df14/data-prepper-plugins/otel-trace-source/src/main/java/org/opensearch/dataprepper/plugins/source/oteltrace/OTelTraceSourceConfig.java#L101-L104

Reply:
Updated the validation. Also noticed that we need `@Valid` on `entries` to make validation take effect in inner fields. Added it to other mutate_event processors where applicable.